src/services/PaymentService.ts

```diff
@@ -282,6 +283,38 @@ export class PaymentService {
       }));
   }
 
+  async getPaypalSetupIntent(
+    priceId: string,
+    user: {
+      name?: string;
+      email: string;
+    },
+  ): Promise<Stripe.SetupIntent> {
+    const getPriceProduct = await this.getPrices();
+    const priceProduct = getPriceProduct.find((price) => price.id === priceId);
```

Comment:
Throw an error if you do not find the priceProduct, so we are sure that we are not receiving fake priceIds